### ISSUE TITLE
Refactor `get_prefill_values` logic for all plugins

### ISSUE THREAD

**Issue by dariomory** [open-formulieren/open-forms#1891]
### Thema / Theme

Plugins

### Omschrijving / Description

**Problem:** Most plugins `get_prefill_values` are handling exceptions silently and returning `{}`.

**Solution:**  Let plugins raise exceptions, `_fetch_prefill_values` handles everything (logevent, logger) in one place. 

https://github.com/open-formulieren/open-forms/blob/24f2870b1557c2671af3f02f5f7814c8b9308961/src/openforms/prefill/__init__.py#L63-L71

**Comment by sergei-maertens** [comment]
This was done on purpose to prevent errors propagating to end-users, so we're not changing that.

**Comment by dariomory** [comment]
@sergei-maertens It is a better practice to raise a custom exception explicitly, like it is done in `check_config`.

See issue: https://github.com/open-formulieren/open-forms/pull/1852

**Comment by sergei-maertens** [comment]
No, this is incorrect. The plugin just has to ensure it doesn't swallow exceptions, and we have the generic exception handler which logs to the database, see: https://github.com/open-formulieren/open-forms/blob/2519ac97b260737d0392c918856d08728e8431c5/src/openforms/prefill/__init__.py#L63

**Comment by sergei-maertens** [comment]
After working on #1842 more closely, yes, plugins should indeed not swallow exceptions and they should be caught by `openforms.prefill._fetch_prefill_values`.

The custom exception part is **not** where we are going, we're sticking to catching bare `Exception` as that protects us against programming mistakes, while logging appropriately and preventing hard HTTP 500 errors for end users.
